caput/mpiarray.py

```diff
@@ -1378,15 +1378,15 @@ def reshape(self, *shape):
 
         return rdata
 
-    def copy(self):
+    def copy(self, *args, **kwargs):
         """Return a copy of the MPIArray.
 
         Returns
         -------
         arr_copy : MPIArray
         """
         return MPIArray.wrap(
-            self.view(np.ndarray).copy(), axis=self.axis, comm=self.comm
+            self.view(np.ndarray).copy(*args, **kwargs), axis=self.axis, comm=self.comm
```

Comment:
As you can guarantee this is properly formed you can probably replace the wrap with `_view_from_data_and_params` which would save some collective communication.


Finally, it *might* be possible to remove this copy method entirely and just reply on using numpys directly. Now we've correctly implemented all the area creation hooks numpys own copy method might just work as-is.

Reply:
I'll give both a try